src/js/components/MosaicView.js

```diff
@@ -84,59 +116,75 @@ MosaicView.propTypes = {
   camera: cameraType,
 }
 
-function ChannelView({ locationName, camera, view, currentMeta }) {
+function ChannelView({
+  locationName,
+  camera,
+  view,
+  currentMeta,
+  setHasFocus,
+  isFocusable,
+}) {
   const channel = camera.channels.find(({ name }) => name === view.channel)
   if (!channel) {
     return <h3>Channel {view.channel} not found</h3>
   }
-  const { latestEvent: { day_obs: dayObs }} = view
+  const {
+    hasFocus,
+    mediaType,
+    latestEvent: { day_obs: dayObs },
+  } = view
+  const clsName = ["view", `view-${mediaType}`, hasFocus ? "has-focus" : null]
+    .join(" ")
+    .trimEnd()
+  const clickHandler = isFocusable ? () => setHasFocus(view) : null
   return (
-    <>
-      <h3 className="channel">{channel.title}
-        { dayObs && (
-          <span>: { dayObs }</span>
-        ) }
+    <li className={clsName} onClick={clickHandler}>
+      <h3 className="channel">
+        {channel.title}
+        {dayObs && <span>: {dayObs}</span>}
       </h3>
       <ChannelMedia
         locationName={locationName}
         camera={camera}
         event={view.latestEvent}
+        mediaType={mediaType}
       />
-      <ChannelMetadata
-        view={view}
-        metadata={currentMeta}
-      />
-    </>
+      <ChannelMetadata view={view} metadata={currentMeta} />
+    </li>
   )
 }
 ChannelView.propTypes = {
   locationName: PropTypes.string,
   camera: cameraType,
   view: mosaicSingleView,
-  currentMeta: metadataType
-}
-
-function ChannelMedia({ locationName, camera, event }) {
-  const { filename, ext } = event
-  const mediaURL = buildMediaURI(locationName, camera.name, event.channel_name, filename)
-  switch (ext) {
-    case 'mp4':
-      return <ChannelVideo mediaURL={mediaURL}/>
-    case 'jpg':
-    case 'jpeg':
-    case 'png':
-      return <ChannelImage mediaURL={mediaURL}/>
+  currentMeta: metadataType,
+}
+
+function ChannelMedia({ locationName, camera, event, mediaType }) {
+  const { filename } = event
+  if (!filename) return <ChannelMediaPlaceholder />
+  const mediaURL = buildMediaURI(
+    locationName,
+    camera.name,
+    event.channel_name,
+    filename
+  )
+  switch (mediaType) {
+    case "video":
```

Comment:
Similar to a comment before, I recommend to define an enum in Javascript (just an object) such as you are doing for the python code on `models.py`.